arcgis-ios-sdk-samples/Edit data/Add features with contingent values/ContingentValuesTableViewController.swift

```diff
@@ -0,0 +1,324 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+protocol ContingentValuesDelegate: AnyObject {
+    func createGraphics(for feature: AGSFeature) -> AGSGraphic
+}
+
+class ContingentValuesTableViewController: UITableViewController {
+    @IBOutlet var statusCell: UITableViewCell!
+    @IBOutlet var protectionCell: UITableViewCell!
+    @IBOutlet var bufferSizeCell: UITableViewCell!
+    @IBOutlet var doneBarButtonItem: UIBarButtonItem!
+    @IBOutlet var bufferSizePickerView: UIPickerView!
+    
+    // MARK: Actions
+    
+    @IBAction func cancelBarButtonItemTapped(_ sender: UIBarButtonItem) {
+        // Remove the last graphic added.
+        graphicsOverlay?.graphics.removeLastObject()
+        // Dismiss the table view.
+        dismiss(animated: true)
+    }
+    
+    @IBAction func doneBarButtonItemTapped(_ sender: Any) {
+        guard let feature = feature, let featureTable = featureTable else { return }
+        // Set the feature's geometry to the map point.
+        feature.geometry = mapPoint
+        // Add the feature to the feature table.
+        featureTable.add(feature) { [unowned self] _ in
+            // Create buffer graphics for the new feature.
+            let graphic = self.delegate?.createGraphics(for: feature)
+            graphicsOverlay?.graphics.add(graphic)
```

Comment:
A few problems here:

1. An async method is blocking the dismiss UI, which means the done button tap event can halt for indefinite time if add feature operation takes longer.
2. The result of the delegate method is used in this `ContingentValuesTableViewController`, which is not what delegate pattern should be.

It would be better like this:

1. Have a delegate method such as `func contingentValuesTableViewController(_ controller: ContingentValuesTableViewController, didFinishWith feature: AGSFeature)`
2. No need to have `createGraphics` as a protocol stub. When the `didFinishWith` method is called, you can call `createGraphics` in it, in the `AddFeaturesContingentValuesViewController`, which means

```swift
extension AddFeaturesContingentValuesViewController: ContingentValuesDelegate {
    func contingentValuesTableViewController(_ controller: ContingentValuesTableViewController, didFinishWith feature: AGSFeature) {
        let graphic = createGraphics(for: feature)
        // Add the feature to the feature table.
        featureTable?.add(feature) { [weak self] _ in
            self?.graphicsOverlay.graphics.add(graphic)
        }
    }
}
```

3. This means when the done button is tapped, only a contingent-value-constrained feature gets passed back to the `AddFeaturesContingentValuesViewController`. The other work can be done afterwards there.